lightning-invoice/src/payment.rs

```diff
@@ -173,10 +173,32 @@ where
 	logger: L,
 	event_handler: E,
 	/// Caches the overall attempts at making a payment, which is updated prior to retrying.
-	payment_cache: Mutex<HashMap<PaymentHash, usize>>,
+	payment_cache: Mutex<HashMap<PaymentHash, PaymentAttempts>>,
 	retry_attempts: RetryAttempts,
 }
 
+/// Storing minimal payment attempts information required for determining if a outbound payment can
+/// be retried.
+#[derive(Debug,Clone,Copy)]
+struct PaymentAttempts {
+    /// This should be >= 1 as we only insert PaymentAttempts for a PaymentHash if there is at least
+    /// one attempt.
+    counts: usize,
+    latest_attempted_at: SystemTime,
+    first_attempted_at: SystemTime
```

Comment:
Note that `SystemTime` is only available with `std` feature, so this will need to be conditional. See places in this file where `#[cfg(feature = "std")]` is used.

With that in mind, `RetryAttempts::MaxDuration` won't be applicable without the feature, too.